### PR TITLE
Datetime delta

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

See discussion at conda-forge/arrow-cpp-feedstock#717, pandas-dev/pandas#50817

### CODE REVIEW

**Comment by h-vetinari** [comment]
This seems to fail based on conda-forge now having switched fully to OpenSSL 3? 🤔 

```
* _ssl_build.py
fetching archive http://artfiles.org/openssl.org/source/openssl-1.1.1s.tar.gz
* _sqlite3_build.py
stdout:
_ssl archive /tmp/pypy-archives/openssl-1.1.1s.tar.gz hash mismatch
```

Once this passes I'd like to enable testing with pypy on the pandas feedstock ([with](https://github.com/conda-forge/pandas-feedstock/pull/162) or without Cython 3)

**Comment by mattip** [comment]
On MacOS, something changed in compilation. The function signature PyPy uses for `pthread_kill` is wrong, but it has been that way for quite a long time.

On linux, as @h-vetinari points out, PyPy is building the cffi imports with [embed_dependencies = True](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=701996&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=2347) which is wrong for conda.

**Comment by mattip** [comment]
CI is passing, this is ready for review.

File: recipe/build.sh [hunk 1]
```diff
@@ -30,10 +30,10 @@ RELEASE_DIR=$PYPY3_SRC_DIR/pypy/tool/release
 PYPY_PKG_NAME=pypy3
 BUILD_DIR=${PREFIX}/../build
 TARGET_DIR=${PREFIX}/../target
-ARCHIVE_NAME="${PYPY_PKG_NAME}-${PKG_VERSION}"
-PYPY_PACKAGE_WITHOUTTK=1
-PYPY_NO_EMBED_DEPENDENCIES=1
-PYPY_NO_MAKE_PORTABLE=1
+export ARCHIVE_NAME="${PYPY_PKG_NAME}-${PKG_VERSION}"
+export PYPY_PACKAGE_WITHOUTTK=1
+export PYPY_NO_EMBED_DEPENDENCIES=1
```
Comment (h-vetinari) [inline recipe/build.sh:35]: If this wasn't working before then this PR seems like a fairly important fix?

**Review by h-vetinari — APPROVED** [review]

File: recipe/build.sh (hunk 1 above)
Comment (mattip) [reply recipe/build.sh:35]: I'm not sure. I can never tell when I am required to use `export` in a bash script that calls other executables. These environment variables are used int these lines
`PYTHONPATH=../.. ./${PYPY_PKG_NAME}*-c ../../lib_pypy/pypy_tools/build_cffi_imports.py`
and 
`${PYTHON} ./package.py --targetdir="$TARGET_DIR" --archive-name="$ARCHIVE_NAME"` which internally use `subprocess.Popen`.

**Comment by mattip** [comment]
> Once this passes I'd like to enable testing https://github.com/conda-forge/pandas-feedstock/pull/162 pypy on the pandas feedstock (with or without Cython 3)

OK. I see testing PyPy is disabled due to segfaults, issue #148, and upstream has also disabled testing PyPy. I wonder if it still segfaults. In any case, testing will be very slow, the JIT can do little with the heavily compiled pandas code.

**Pull request merged**